askama_derive/src/input.rs

```diff
@@ -113,32 +113,63 @@ impl TemplateInput<'_> {
+                            let extends = self.config.find_template(extends.path, Some(&path))?;
+                            let dependency_path = (path.clone(), extends.clone());
+                            if dependency_graph.contains(&dependency_path) {
+                                return Err(format!(
+                                    "cyclic dependency in graph {:#?}",
+                                    dependency_graph
+                                        .iter()
+                                        .map(|e| format!("{:#?} --> {:#?}", e.0, e.1))
+                                        .collect::<Vec<String>>()
+                                )
+                                .into());
+                            }
+                            dependency_graph.push(dependency_path);
+                            let source = get_template_source(&extends)?;
+                            check.push((extends, source));
                         }
-                        dependency_graph.push(dependency_path);
-                        let source = get_template_source(&extends)?;
-                        check.push((extends, source));
-                    }
-                    Node::Import(import) => {
-                        let import = self.config.find_template(import.path, Some(&path))?;
-                        let source = get_template_source(&import)?;
-                        check.push((import, source));
+                        Node::Macro(m) if top => {
+                            nested.push(&m.nodes);
+                        }
+                        Node::Import(import) if top => {
+                            let import = self.config.find_template(import.path, Some(&path))?;
+                            let source = get_template_source(&import)?;
+                            check.push((import, source));
+                        }
+                        Node::Include(include) => {
+                            let include = self.config.find_template(include.path, Some(&path))?;
+                            let source = get_template_source(&include)?;
+                            check.push((include, source));
+                        }
+                        Node::BlockDef(b) => {
+                            nested.push(&b.nodes);
+                        }
+                        Node::If(i) => {
+                            for cond in &i.branches {
+                                nested.push(&cond.nodes);
+                            }
+                        }
+                        Node::Loop(l) => {
+                            nested.push(&l.body);
+                            nested.push(&l.else_nodes);
+                        }
+                        Node::Match(m) => {
+                            for arm in &m.arms {
+                                nested.push(&arm.nodes);
+                            }
+                        }
+                        _ => {}
```

Comment:
Since this is now listing many `Node` variants, I'd prefer to explicitly list variants to be ignored here instead of allowing the rest to silently match. That way, we'll explicitly have to consider whether a new variant needs to be handled here.

Reply:
The ignored node list is quite long but okay

Reply:
I suggest doing a `use Node::*` before the `match` to avoid the repetition.

Reply:
done